service/resource/legacyv2/resource.go

```diff
@@ -207,12 +215,31 @@ func (s *Resource) NewUpdatePatch(ctx context.Context, obj, currentState, desire
 
 	s.logger.Log("info", fmt.Sprintf("cluster '%s' processed", keyv2.ClusterID(cluster)))
 
-	return &framework.Patch{}, nil
+	patch := framework.NewPatch()
+
+	// cloudformation logic
```

Comment:
I think more comments here would be good. To mention that the resources are combined during the transition due to the deletion order problem and that they will be separated later.

Reply:
Indeed, I've extended the existing ones, please take a look.